File: C#/118. Pascal's Triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        List<IList<int>> arr= new List<IList<int>>();
        for(int i = 0; i<numRows;i++){
            var row = new List<int>();
            for (int j = 0; j<=i;j++){
                if((j-1)<0 || (j+1)>i){
                    row.Add(1);
                }
                else{
                    row.Add(arr[i-1][j-1] + arr[i-1][j]);
                }
            }
            arr.Add(row);
        }
        return arr;
    }
}

File: Java/118. Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        int[][] arr = new int[numRows][];
        for(int i = 0; i<numRows;i++){
            int[] row = new int[i + 1];
            row[0] = 1;
            row[i] = 1;
            for (int j = 1; j<i;j++){
                row[j]=arr[i-1][j-1]+arr[i-1][j];
            }
            arr[i] = row;
        }
        return (List)Arrays.asList(arr);
    }
}
